src/pal/src/thread/process.cpp

```diff
@@ -1990,14 +2048,90 @@ PAL_NotifyRuntimeStarted()
     return launched;
 }
 
-#ifdef __APPLE__
 LPCSTR
 PALAPI
 PAL_GetApplicationGroupId()
 {
+#ifdef __APPLE__
     return gApplicationGroupId;
+#else // __APPLE
+    return nullptr;
+#endif
+}
+
+#ifdef __APPLE__
+
+// We use 7bits from each byte
+constexpr int BitNum2ByteNum(UINT number)
+{
+    return (number+6)/7;
```

Comment:
>(number+6) [](start = 11, length = 10)

I am a little confused by this. Did you mean to multiply `number` by 8 somewhere?

Reply:
Oh, I see, its because it is used to compute the extra length. I found this confusing. My suggestion:

```
// We use 7bits from each byte
constexpr int GetEncodedByteCount(UINT rawByteCount)
{
    return rawByteCount + (rawByteCount+6)/7;
}
const int SEMAPHORE_ENCODED_NAME_LENGTH =GetEncodedByteCount(sizeof(UnambiguousProcessDescriptor));
```

---
In reply to: [234724292](https://github.com/dotnet/coreclr/pull/21068#discussion_r234724292) [](ancestors = 234724292)